### ISSUE TITLE
Ranger sign inversion

### ISSUE THREAD

**Issue by i404788** [kozistr/pytorch_optimizer#232]
## Describe the bug

From my experiments it seems like the sign for the Ranger is inverted. All other optimizers (including Ranger21) has steps in the opposite direction of Ranger. 

Note that I'm testing context-free step directions/magnitudes using a 'perfect' gradient (scaled by 4), so if Ranger somehow reverts course when gradients from different directions are accumulated that would be missed from my test.
Hyperparameters: `{'betas': (0.003344506587403595, 0.9685357345548955), 'lr': 0.4616639698903086}` (found through hyperparameter search, also done for the other optimizers) and evaluated on the Ackley (dim=2) function.

(I didn't want to create a PR before discussing if this might be intended)

## To Reproduce

* OS : Linux
* PyTorch version : 2
* Python version :  3.11

## Log

Ranger:
![image](https://github.com/kozistr/pytorch_optimizer/assets/50617709/e607fff8-2264-4dfc-bab2-1293d0776837)

For comparison SGD:
![image](https://github.com/kozistr/pytorch_optimizer/assets/50617709/69e25fc7-3456-489e-bec2-7c7c2fdee7a1)

**Comment by kozistr** [comment]
hi! sorry for the late reply

the current Ranger implementation in this repository is based on [here](https://github.com/lessw2020/Ranger-Deep-Learning-Optimizer/blob/master/ranger/ranger2020.py).

(I might be wrong) I guess there's no big difference between the original implementation and mine. (let me know if I'm wrong)

could you please test with the original implementation and get the same result?

thank you!

**Comment by kozistr** [comment]
@i404788 Hi, sorry for late reply. I just added `Ackley` function (dim=2, a=20, b=0.2, c=2pi) to `examples/visualize_optimizers.py`, and I tested with your hyperparameters (betas = (0.003344506587403595, 0.9685357345548955), leave `lr` for tuning).

here's my test result (best loss is at lr = 0.71 in my case, initial point (-2, -2)). it looks like it converged to the global minimum well while the direction is changed in the middle of steps.

![Image](https://github.com/user-attachments/assets/efd2db35-08ec-4618-b990-870f3800f936)

if you are still interested in this topic, could you elaborate more on this? thank you!

**Comment by i404788** [comment]
Hmm I tried re-running it with your hyperparameters, but still have divergence on the initial step with v3.5.0
As mentioned in the original post, this could be intended, but given your experiment it seems like it would invert again in later steps.

Ranger:
![Image](https://github.com/user-attachments/assets/ddffb8c4-0859-418b-ab53-c42f9f0717fa)

SGD:

![Image](https://github.com/user-attachments/assets/fa75c70e-9fe3-47af-8e95-bde1713dcf5a)

**Comment by kozistr** [comment]
@i404788 Hi. I'm going to close this issue (it looks like it might be intended). If you have an idea or want to propose something, please feel free to reopen this issue or open a PR. Thanks!
